src/tracks/tracks.c

```diff
@@ -42,44 +62,60 @@ const Tracks * get_tracks()
 }
 
 
-int add_track(const Track *track, size_t index, int mode)
+enum track_add_result add_track(const Track *track, const size_t index,
+                                const enum track_add_mode mode)
 {
-    int result = TRACK_ADD_RESULT_OK;
+        if (index >= MAX_TRACK_COUNT) {
+                pr_error("Invalid track index\r\n");
+                return TRACK_ADD_RESULT_FAIL;
+        }
+
+        switch (mode) {
+        case TRACK_ADD_MODE_IN_PROGRESS:
+        case TRACK_ADD_MODE_COMPLETE:
+                /* Valid cases.  Carry on */
+                break;
+        default:
+                pr_error_int_msg("Unknown track_add_mode: ", mode);
+                return TRACK_ADD_RESULT_FAIL;
+        }
+
+        static Tracks *g_tracksBuffer;
+        if (g_tracksBuffer == NULL) {
+                terminate_lua();
```

Comment:
Ok.  I'll buy that argument.

On Fri, Oct 2, 2015 at 1:28 PM, Brent Picasso <notifications@github.com>
wrote:

> In src/tracks/tracks.c
> <https://github.com/autosportlabs/RaceCapture-Pro_firmware/pull/92#discussion_r41052211>
> :
>
> > +                return TRACK_ADD_RESULT_FAIL;
> > +        }
> > +
> > +        switch (mode) {
> > +        case TRACK_ADD_MODE_IN_PROGRESS:
> > +        case TRACK_ADD_MODE_COMPLETE:
> > +                /* Valid cases.  Carry on */
> > +                break;
> > +        default:
> > +                pr_error_int_msg("Unknown track_add_mode: ", mode);
> > +                return TRACK_ADD_RESULT_FAIL;
> > +        }
> > +
> > +        static Tracks *g_tracksBuffer;
> > +        if (g_tracksBuffer == NULL) {
> > +                terminate_lua();
>
> We're now limiting Lua RAM usage so this shouldn't be a problem for MK2.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/autosportlabs/RaceCapture-Pro_firmware/pull/92/files#r41052211>
> .
>



--